### PR TITLE
knot-dns: 3.2.0 -> 3.2.1

### PR DESCRIPTION
**Pull request by vcunat**
https://gitlab.nic.cz/knot/knot-dns/-/tags/v3.2.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
BTW, now it's ready for libbpf 1.0 (PR #187978), though libxdp has to be added first (PR #185559).

**Comment by vcunat** [comment]
@ofBorg build knot-dns.tests

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/servers/dns/knot-dns/default.nix [hunk 1]
```diff
@@ -63,6 +58,19 @@ stdenv.mkDerivation rec {
...
+    deps = runCommandLocal "knot-deps-test"
+      { nativeBuildInputs = [ (lib.getBin stdenv.cc.libc) ]; }
+      ''
+        for libname in libngtcp2 ${lib.optionalString stdenv.isLinux "libbpf"}; do
+          echo "Checking for $libname:"
+          ldd '${knot-dns.bin}/bin/knotd' | grep -F "$libname"
+          echo "OK"
+        done
+        touch "$out"
+      '';
```
Comment (SuperSandro2000) [inline pkgs/servers/dns/knot-dns/default.nix:73]: I am not really convinced on the usefulness of this.

Comment (vcunat) [reply pkgs/servers/dns/knot-dns/default.nix:73]: I don't feel strongly about that.  I wasn't sure, but I'd do similar checks manually on updates of `knot-dns` or these dependencies, so I thought why not automate them (ofBorg) and also make it easier for me (`nix build -f. knot-dns.tests`).  I don't expect they'd have... noticeable cons, even if the pros are minor.

**Comment by vcunat** [comment]
This force-push is just moving `knot-dns.tests.deps` to linux-only section.  I didn't realize that `ldd` is missing on darwin and I don't know a good replacement.  (That also collapsed a conditional on linux-only deps in there.)

**Pull request merged**